File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabble

import "strings"

type TScrabble struct {
    Value int
    Letters string
}

func Score(word string) int {
     scrabbleValues := []TScrabble{
        { Value: 1, Letters: "AEIOULNRST"},
        { Value: 2, Letters: "DG"},
        { Value: 3, Letters: "BCMP"},
        { Value: 4, Letters: "FHVWY"},
        { Value: 5, Letters: "K"},
        { Value: 8, Letters: "JX"},
        { Value: 10, Letters: "QZ"},
    }   
    
	var score int
    for _, letter := range word {
        for _, sv := range scrabbleValues {
            if strings.Contains(sv.Letters, strings.ToUpper(string(letter))) {
                score += sv.Value
            }
        }
    }
    return score
}

File: solutions/csharp/scrabble-score/1/ScrabbleScore.cs
public static class ScrabbleScore
{
    public static int Score(string input)
    {
        var scrabble = new Dictionary<char[], int>
        {
            { new char[] {'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'}, 1 },
            { new char[] {'D', 'G'}, 2 },
            { new char[] {'B', 'C', 'M', 'P'}, 3 },
            { new char[] {'F', 'H', 'V', 'W', 'Y'}, 4 },
            { new char[] {'K'}, 5 },
            { new char[] {'J', 'X'}, 8 },
            { new char[] {'Q', 'Z'}, 10 },
        };
        var score = 0;
        foreach (var letter in input.ToUpper())
        {
            foreach (var (key, value) in scrabble)
            {
                if (key.Contains(letter))
                {
                    score += value;
                    break;
                }
            }
        }
        return score;
    }
}